tests/pypi/extension/extension_tests.bzl

```diff
@@ -403,6 +403,90 @@ new-package==0.0.1 --hash=sha256:deadb00f2
+                        hub_name = "pypi",
+                        python_version = "3.12",
+                        download_only = True,
+                        requirements_by_platform = {
+                            "requirements.linux_arm64.txt": "linux_aarch64",
+                            "requirements.linux_x86_64.txt": "linux_x86_64",
+                        },
+                        experimental_index_url = "pypi.org",
+                    ),
+                ],
+            ),
+            read = lambda x: {
+                "requirements.linux_arm64.txt": """\
+package==0.7.0 \
+    --hash=sha256:4dd8924f171ed73a4f1a6191e2f800ae1745069989b69fabc45593d6b6504003 \
+    --hash=sha256:62833036cbaf4641d66ae94c61c0446890a91b2c0d153946583a0ebe04877a76
+""",
+                "requirements.linux_x86_64.txt": """\
+package[extra]==0.7.0 \
+    --hash=sha256:62833036cbaf4641d66ae94c61c0446890a91b2c0d153946583a0ebe04877a76
+""",
+            }[x],
+        ),
+        available_interpreters = {
+            "python_3_12_host": "unit_test_interpreter_target",
+        },
+        minor_mapping = {"3.12": "3.12.11"},
+        simpleapi_download = lambda *_, **__: {
+            "package": parse_simpleapi_html(
+                url = "https://example.com/package",
+                content = """
+<a href="package-0.7.0.tar.gz#sha256=4dd8924f171ed73a4f1a6191e2f800ae1745069989b69fabc45593d6b6504003">package-0.7.0.tar.gz</a>
+<a href="package-0.7.0-py3-none-any.whl#sha256=62833036cbaf4641d66ae94c61c0446890a91b2c0d153946583a0ebe04877a76">package-0.7.0-py3-none-any.whl</a>
+""",
+            ),
+        },
+    )
+
+    pypi.exposed_packages().contains_exactly({"pypi": ["package"]})
+    # TODO(hartikainen): Check these expectations.
+    pypi.hub_whl_map().contains_exactly({"pypi": {
+        "package": {
+            "pypi_312_package_py3_none_any_62833036": [
+                whl_config_setting(
+                    # TODO(hartikainen): I think all these platforms use the same `.whl`
+                    # and thus all three platforms should be included in the same
+                    # `target_platforms` here?
+                    target_platforms = ["cp312_linux_arm64", "cp312_linux_x86_64"],
+                    version = "3.12",
+                ),
+            ],
+        },
+    }})
+    pypi.whl_libraries().contains_exactly({
+        "pypi_312_package_py3_none_any_62833036": {
+            "dep_template": "@pypi//{name}:{target}",
+            "download_only": True,
+            "experimental_target_platforms": ["linux_arm64", "linux_x86_64"],
+            "filename": "package-0.7.0-py3-none-any.whl",
+            "python_interpreter_target": "unit_test_interpreter_target",
+            # NOTE(hartikainen): Perhaps this is part of the problem?
+            # This should say `package[extra]==0.7.0` for `linux_x86_64` platform and
+            # `package==0.7.0` for `linux_arm64`
+            "requirement": "package[extra]==0.7.0",
```

Comment:
Hah, actually this is a separate issue and I am not sure how we can fix it. So, a little bit of history here:
1. poetry will export a `requirements.txt` that will have both `package[extra]` and `package` and we were asked to only consider `package[extra]` when doing the installation.
2. However with the advent of `universal` files it means that we should do something differently - first resolve the compatible lines with the target platforms and do the filtering differently.

I remember we had this discussion with in one of @ewianda  PRs, but I can't remember which one it is.

I think the issue that you are facing does not have an open issue yet.

Reply:
What is more, we have a test that already tests for this behaviour in the case where the extras differ, but we don't have a test where extra is present in one and in another it is not.

PRs for this are welcome and we should have enough test coverage in places that the fix would be touching.